ncdfchecker.py

```diff
@@ -189,16 +193,66 @@ def check_globals(product, constraints, skip=["short_name"], strict=False,
     return errcount, warncount
 
 
-def check_stepsize(data, stepsize):
+def get_month_midpoints(monthslist):
+    """Return a list of dates that correspond to the middle of the month."""
+    def midpoint(start, end): return start + (end - start) / 2
+
+    midmonths = [midpoint(monthslist[i], monthslist[i+1])
+                 for i in range(len(monthslist)-1)]
+
+    return midmonths
+
+
+def get_expected_monthly_stepsize(initdate, runlength):
     """
-    Helper routine to check that the stepsize in some given data is equal to
-    a given value
+    Using the model initialisation date and run length, calculate the
```

Comment:
No models here - this is purely data driven with no knowledge of source, be it model, obs or otherwise.

Reply:
Here, as elsewhere, I have removed any need of relying on externally specifying the start date and model run length.